catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -92,39 +113,187 @@ public void initialize(Map<String, String> config, CatalogInfo info) throws Runt
+    if (!schemaExists(schemaIdent)) {
+      LOG.warn("Kafka catalog schema {} does not exist", schemaIdent);
+      throw new NoSuchSchemaException("Schema %s does not exist", schemaIdent);
+    }
+
+    try {
+      ListTopicsResult result = adminClient.listTopics();
+      Set<String> topicNames = result.names().get();
+      return topicNames.stream()
+          .map(name -> NameIdentifier.of(namespace, name))
+          .toArray(NameIdentifier[]::new);
+    } catch (InterruptedException | ExecutionException e) {
+      throw new RuntimeException("Failed to list topics under namespace " + namespace, e);
+    }
   }
 
   @Override
   public Topic loadTopic(NameIdentifier ident) throws NoSuchTopicException {
-    throw new UnsupportedOperationException();
+    DescribeTopicsResult result = adminClient.describeTopics(Collections.singleton(ident.name()));
+    int partitions;
+    int replicationFactor;
+    try {
+      TopicDescription topicDescription = result.topicNameValues().get(ident.name()).get();
+      partitions = topicDescription.partitions().size();
+      replicationFactor = topicDescription.partitions().get(0).replicas().size();
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof org.apache.kafka.common.errors.UnknownTopicOrPartitionException) {
+        throw new NoSuchTopicException(e, "Topic %s does not exist", ident);
+      } else {
+        throw new RuntimeException("Failed to load topic " + ident.name() + " from Kafka", e);
+      }
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Failed to load topic " + ident.name() + " from Kafka", e);
+    }
+
+    LOG.info("Loaded topic {} from Kafka", ident);
+
+    return KafkaTopic.builder()
+        .withName(ident.name())
+        .withProperties(
+            ImmutableMap.of(
+                KafkaTopicPropertiesMetadata.PARTITION_COUNT, String.valueOf(partitions),
+                KafkaTopicPropertiesMetadata.REPLICATION_FACTOR, String.valueOf(replicationFactor)))
+        .withAuditInfo(
+            AuditInfo.builder()
+                .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                .withCreateTime(Instant.now())
+                .build())
+        .build();
   }
 
   @Override
   public Topic createTopic(
       NameIdentifier ident, String comment, DataLayout dataLayout, Map<String, String> properties)
       throws NoSuchSchemaException, TopicAlreadyExistsException {
-    throw new UnsupportedOperationException();
+    NameIdentifier schemaIdent = NameIdentifier.of(ident.namespace().levels());
+    if (!schemaExists(schemaIdent)) {
+      LOG.warn("Kafka catalog schema {} does not exist", schemaIdent);
+      throw new NoSuchSchemaException("Schema %s does not exist", schemaIdent);
+    }
+
+    try {
+      CreateTopicsResult createTopicsResult =
+          adminClient.createTopics(Collections.singleton(buildNewTopic(ident, properties)));
```

Comment:
replicaAssignments is more useful in production. Can you consider supporting it?

Reply:
What I'm thinking is that replicaAssignments is a low-frequency and complex operation, so I do not intend to support it in this PR. Instead, I will evaluate whether to add it based on user feedback after implementation.